### PR TITLE
New play board added, client class refactor

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/connection/Client.java [hunk 1]
```diff
@@ -1,35 +1,64 @@
...
+import java.util.Scanner;
 
 public class Client {
 
-    Socket socket;
-    BufferedWriter writer;
-    BufferedReader reader;
+    static final String CHARSET = "UTF-8";
+
+    Connector connector;
```
Comment (M-Aarsman) [inline client/src/main/java/connection/Client.java]: why not private?

File: client/src/main/java/connection/Client.java [hunk 2]
```diff
@@ -1,35 +1,64 @@
...
     public void setup(String host, String port) {
         try {
-            socket = new Socket(host, Integer.valueOf(port));
-            writer = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream(), "UTF-8"));
-            reader = new BufferedReader(new InputStreamReader(socket.getInputStream(), "UTF-8"));
+            connector = SocketConnector.from(host, Integer.valueOf(port));
+            in = MessageIn.from(connector);
+            out = MessageOut.from(connector);
         } catch (IOException e) {
             e.printStackTrace();
```
Comment (M-Aarsman) [inline client/src/main/java/connection/Client.java]: add logger in the future. Only printStackTarce no action needed? If after setup other function will use connector, in or out will not know which of them is invalid

File: client/src/main/java/connection/Client.java [hunk 3]
```diff
@@ -1,35 +1,64 @@
...
-        socket.close();
+    private static boolean loopCondition = true;
+
+    public static void run() throws IOException {
+
+        System.out.println("Welcome to the battleships game, please provide your name");
+        Scanner scan = new Scanner(System.in);
+        String name = scan.nextLine();
+
+        Connector connector = SocketConnector.from("localhost", 5000);
```
Comment (M-Aarsman) [inline client/src/main/java/connection/Client.java]: magic values

File: client/src/main/java/connection/MessageIn.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package connection;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Scanner;
+
+public class MessageIn implements Receiver {
+
+    BufferedReader scanner;
```
Comment (M-Aarsman) [inline client/src/main/java/connection/MessageIn.java]: why not private final?

File: client/src/main/java/connection/MessageIn.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    private MessageIn(BufferedReader scanner) {
+        this.scanner = scanner;
+    }
+
+    @Override
+    public String readMessage() {
+        try {
+            return scanner.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (M-Aarsman) [inline client/src/main/java/connection/MessageIn.java]: use logger in the future

File: client/src/main/java/connection/SocketConnector.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package connection;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class SocketConnector implements Connector{
+
+    Socket socket;
```
Comment (M-Aarsman) [inline client/src/main/java/connection/SocketConnector.java]: why not private final

File: client/src/main/java/playing/FieldPosition.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package playing;
+
+public class FieldPosition {
+
+
+    private final Integer x;
+    private final Integer y;
+
+    FieldPosition(double x, double y) {
+        this.x = (int) x;
```
Comment (M-Aarsman) [inline client/src/main/java/playing/FieldPosition.java]: why? if you store Integer why constructor has doubles?

File: client/src/main/java/playing/PlayBoardController.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    private Client client;
+
+    @FXML
+    GridPane shipBoard;
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        client = new Client();
+        client.setup("localhost", "5000");
```
Comment (M-Aarsman) [inline client/src/main/java/playing/PlayBoardController.java]: magic values

File: client/src/main/java/playing/SeaField.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package playing;
+
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+
+
+public class SeaField extends Rectangle implements Field {
+
+    SeaField(int i, int n) {
```
Comment (M-Aarsman) [inline client/src/main/java/playing/SeaField.java]: maybe better names then i an n please?

File: client/src/main/java/starting/StartBoard.java [hunk 10]
```diff
@@ -2,31 +2,49 @@
...
 
-        StartBoardController startBoardController = loader.getController();
-        System.out.println(startBoardController.ipNumber.getText()); //TODO: change to Logger
+        FXMLLoader startLoader = new FXMLLoader(getClass().getResource(START_BOARD_URL));
+        AnchorPane startBoard = startLoader.load();
+        addNextButtonToStartBoard(stage, playScene, startBoard);
+        startRoot.getChildren().addAll(startBoard);
+
+        FXMLLoader playLoader = new FXMLLoader(getClass().getResource(PLAY_BOARD_URL));
+        AnchorPane playBoard = playLoader.load(); //TODO to uruchamia drugi wątek!!!!
```
Comment (M-Aarsman) [inline client/src/main/java/starting/StartBoard.java]: TODO why in polish?

File: client/src/main/java/playing/FieldPosition.java (hunk 7 above)
Comment (patiem) [reply client/src/main/java/playing/FieldPosition.java]: Becouse x and y that is passed to constructor is from Gui Rectangle class that stores this values ass doubles

File: client/src/main/java/starting/StartBoard.java (hunk 10 above)
Comment (patiem) [reply client/src/main/java/starting/StartBoard.java]: Because it was 2 o'clock in the middle of the night

**Pull request merged**